app/models/veteran.rb

```diff
@@ -37,6 +37,8 @@ class Veteran < CaseflowRecord
 
   delegate :full_address, to: :address
 
+  before_save :set_death_reported_on_date!
```

Comment:
IMO, it's easier to determine relevance of this hook if you check `if date_of_death_changed?` here.
```suggestion
  before_save :set_death_reported_on_date! if date_of_death_changed?
```